compiler/src/main/scala/org/scalajs/nscplugin/PrepJSExports.scala

```diff
@@ -238,8 +163,13 @@ trait PrepJSExports[G <: Global with Singleton] { this: PrepJSInterop[G] =>
         Nil
     }
 
+    val allAnnots = directAnnots ++ unitAnnots
+
+    if (allAnnots.nonEmpty)
+      checkExportTarget(sym, allAnnots.head.pos)
```

Comment:
An original version had this. What I found is that the resulting code ended up being (IMHO) fairly scalac un-idiomatic (returning booleans for errors). Notably this was IMO problematic because it would spread to other parts of the codebase (`checkSetterSignature`).

> can crash 

This is indeed a problem. But my gut feeling is, that this will only happen when invoking `typer.typedDefDef`. Maybe we should work towards removing that (and type ourselves).

> produce nonsensical Tree

That's not a problem, is it? Because if this phase emits an error, the next one would not run.